src/Acheve.TestHost/Extensions/HttpResponseMessageExtensions.cs

```diff
@@ -0,0 +1,47 @@
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace System.Net.Http;
+
+public static class HttpResponseMessageExtensions
+{
+    /// <summary>
+    /// Try to extract the error message in the response content in case the response status code is not success.
+    /// </summary>
+    /// <param name="response">The HttpResponseMessage instance</param>
+    /// <returns></returns>
+    public static async Task IsSuccessStatusCodeOrThrow(this HttpResponseMessage response)
```

Comment:
HttpResponseMessage already contains a similar method for this purpose ( i think without the content)